modules/commerce_bluesnap_currency/src/EventSubscriber/ExchangeRateBluesnap.php

```diff
@@ -46,24 +54,30 @@ public static function sourceId() {
   public function getExternalData($base_currency = NULL) {
     $external_data = [];
     // Prepare for client.
-    $url = self::apiUrl($this->configFactory->get('bluesnap')['mode']);
+    $url = self::apiUrl(
+      $this->config
+        ->get('bluesnap')['mode']
+    );
     $method = 'GET';
     $options['auth'] = [
-      $this->configFactory->get('bluesnap')['username'],
-      $this->configFactory->get('bluesnap')['password'],
+      $this->config
+        ->get('bluesnap')['username'],
+      $this->config
+        ->get('bluesnap')['password'],
     ];
     $options['headers'] = [
       'Accept' => 'application/json',
     ];
 
     if ($this->apiClient($method, $url, $options)) {
       $response = Json::decode($this->apiClient($method, $url, $options));
-
-      // Loop through the api response and build the exchange rate array.
-      foreach ($response->currencyRate as $rate) {
-        $code = (string) $rate->quoteCurrency;
-        $rate = (string) $rate->conversionRate;
-        $external_data[$code] = $rate;
+      if (!empty($response['currencyRate'])) {
+        // Loop through the api response and build the exchange rate array.
+        foreach ($response['currencyRate'] as $rate) {
+          $code = (string) $rate['quoteCurrency'];
+          $rate = (string) $rate['conversionRate'];
+          $external_data[$code] = $rate;
+        }
```

Comment:
@safalliajoseph In which case would we not have the rates in the response?

Reply:
If for some reason the curl request failed, then we will not have rate in response right? It happened to me once. That is why I added a condition there

Reply:
**CHANGE REQUEST**

As discussed inject the logger and log an error when this happen.
Also, avoid the triple nesting here, it's unnecessary.

Reply:
@krystalcode Made the change. Please review